### PR TITLE
Fixes to tx sub

### PR DESCRIPTION
**Pull request by marta-lokhova**
Update transaction/tx set logic to compute fee before it's needed. Just in case, initialize the fee to 0 to avoid crashing (follow-up to https://github.com/stellar/stellar-core/pull/3730)

### CODE REVIEW

File: src/herder/TxSetFrame.cpp [hunk 1]
```diff
@@ -831,15 +830,30 @@ TxSetFrame::isGeneralizedTxSet() const
...
 {
     size_t oldSize = mTxs.size();
     mTxs.reserve(oldSize + txs.size());
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+    LedgerTxn ltx(app.getLedgerTxnRoot(), false,
+                  TransactionMode::READ_ONLY_WITHOUT_SQL_TXN);
+    auto ledgerVersion = ltx.loadHeader().current().ledgerVersion;
+    auto const& sorobanConfig =
+        app.getLedgerManager().getSorobanNetworkConfig(ltx);
+    ltx.rollback();
```
Comment (sisuresh) [inline src/herder/TxSetFrame.cpp:845]: What's the point of this `rollback`?

File: src/main/main.cpp [hunk 2]
```diff
@@ -263,7 +263,7 @@ main(int argc, char* const* argv)
     checkStellarCoreMajorVersionProtocolIdentity();
     rust_bridge::check_lockfile_has_expected_dep_trees(
         Config::CURRENT_LEDGER_PROTOCOL_VERSION);
-    checkXDRFileIdentity();
+    // checkXDRFileIdentity();
```
Comment (sisuresh) [inline src/main/main.cpp:266]: This should be uncommented.

File: src/herder/TxSetFrame.cpp (hunk 1 above)
Comment (dmkozh) [reply src/herder/TxSetFrame.cpp]: Just wanted to get rid of ltx, but it's probably safe to let it live until the method is finished.

File: src/main/main.cpp (hunk 2 above)
Comment (dmkozh) [reply src/main/main.cpp]: Thanks, updated.

**Comment by sisuresh** [comment]
r+ 9040e2a36b4b87cfd4cde7ccde8ab20882ccf1d9

**Pull request merged**